ext/drutil/drutil.c

```diff
@@ -422,6 +422,11 @@ drutil_insert_get_mem_addr_arm(void *drcontext, instrlist_t *bb, instr_t *where,
             disp = -disp;
             negated = !negated;
         }
+#    ifdef AARCH64
+        /* FIXME i#5498 */
+        if (index == DR_REG_W28)
+            return false;
```

Comment:
Mixed feelings here:

- On the one hand, this seems the right thing to do, putting this in ASAP rather than having silently incorrect instrumentation.
- On the other hand, drmemtrace turns this false return value into a fatal error that kills the process -- and after some analysis last week we found that the highest % of addresses with a w28 index never got above 0.08% in any of our traces, which for many analysis purposes (aggregate statistics over various simulations, e.g.) we can live with more easily than we can live with the tracer just dying on automated tests and jobs where it used to successfully execute.

Is the real fix complex?  At first glance it seems like it may be quite simple?  Expand the index to pointer-sized before comparing; shrink the return value of replace_stolen_reg()?  A 2-line change?  I didn't spend very long looking at it so maybe I missed something.  If it is simple we might prefer to go straight to the fix.

Reply:
> Mixed feelings here...

Same here, I just put this in as a placeholder and a check to see which of our tests regressed.

> after some analysis last week we found that the highest % of addresses with a w28 index never got above 0.08% in any of our traces

That's a relief. I didn't think the % would be high, but that's much lower than I'd hoped for.

> Is the real fix complex? At first glance it seems like it may be quite simple?

I've put a fix in now. I was concerned about violating transparency but after some thought it's not a problem.